public/pages/MonitorDetails/containers/MonitorDetails.js

```diff
@@ -174,14 +178,189 @@ export default class MonitorDetails extends Component {
+  };
+
+  decideWhichFlyout = (validationResponse, adConfigs, queriesForOverview) => {
+    const setFlyout = this.props.setFlyout;
+    if (validationResponse.failures.others) {
+      let message = '';
+      for (let [key, value] of Object.entries(validationResponse.failures)) {
+        if (key === 'others') {
+          message = value;
+        }
+      }
+      this.props.setFlyout({ type: 'detectorFailure', payload: { setFlyout, message } });
+    } else {
+      this.renderFlyout(adConfigs, validationResponse, queriesForOverview);
+    }
+  };
+
+  renderStartedDetectorFlyout = (configs, detectorID, queries) => {
+    const { httpClient } = this.props;
+    const setFlyout = this.props.setFlyout;
+    let startedDetector = true;
+    const adConfigs = configs;
+    const queriesForOverview = queries;
+    this.props.setFlyout({
+      type: 'createDetector',
+      payload: {
+        adConfigs,
+        queriesForOverview,
+        httpClient,
+        setFlyout,
+        startedDetector,
+        detectorID,
+      },
+    });
+  };
+
+  renderFlyout = (adConfigs, validationResponse, queriesForOverview) => {
+    const { httpClient } = this.props;
+    const setFlyout = this.props.setFlyout;
+    const renderStartedDetectorFlyout = this.renderStartedDetectorFlyout;
+    const renderFlyout = this.renderFlyout;
+    const failures = Object.assign(validationResponse.failures);
+    const suggestedChanges = Object.assign(validationResponse.suggestedChanges);
+    const startedDetector = false;
+    this.props.setFlyout({
+      type: 'createDetector',
+      payload: {
+        adConfigs,
+        queriesForOverview,
+        httpClient,
+        setFlyout,
+        renderStartedDetectorFlyout,
+        failures,
+        suggestedChanges,
+        renderFlyout,
+        startedDetector,
+      },
+    });
+  };
+
+  validateADConfigs = async (configs) => {
+    const { httpClient } = this.props;
+    try {
+      const response = await httpClient.post('../api/alerting/detectors/_validate', {
+        configs,
+      });
+      let resp = _.get(response, 'data.response');
+      return resp;
+    } catch (err) {
+      if (typeof err === 'string') throw err;
+      throw 'There was a problem validating the configurations';
+    }
+  };
+
+  getLatestTimeStamp = async (adTimeField, adIndices) => {
+    const { httpClient } = this.props;
+    const searchQuery = {
+      size: 1,
+      sort: [
+        {
+          timestamp: {
+            order: 'desc',
+          },
+        },
+      ],
+      aggregations: {
+        max_timefield: {
+          max: {
+            field: adTimeField,
+          },
+        },
+      },
+    };
+    try {
+      const options = {
+        index: adIndices,
+        query: searchQuery,
+      };
+      const response = await httpClient.post('../api/alerting/_search', options);
+      let maxStamp = _.get(response, 'data.resp.aggregations.max_timefield.value');
```

Comment:
What will the `maxStamp` be if no timestamp is found (empty data)? Will this throw an exception, or just return some default value like 0?

Reply:
Currently the visual graph type monitors require a timestamp to be created so there will always be one while converting from a visual graph monitor. I could query the index mapping for time field for extra protection and especially when other monitor types are converted.

Reply:
Got it, I'm just wondering if no timestamp is found (even if the time field exists) what the query will return. Not sure if ES returns 0 or null on max aggs. Maybe there will already be issues w/ monitor creation if this is the case.